test/e2e/app-dir/use-params/tsconfig.json

```diff
@@ -2,7 +2,7 @@
   "compilerOptions": {
     "lib": ["dom", "dom.iterable", "esnext"],
     "allowJs": true,
-    "skipLibCheck": true,
+    "skipLibCheck": false,
```

Comment:
This is ignored in the e2e test as far as I can tell. The `tsconfig` in the generated, temporary test folder uses `skipLibCheck: true`

Reply:
Hm, it does fail to build for me with the first commit, also in isolation. How did you check this?

```
❯ pnpm testonly-start test/e2e/app-dir/use-params

> nextjs-project@0.0.0 testonly-start <snip>/next.js
> cross-env NEXT_TEST_MODE=start pnpm testonly "test/e2e/app-dir/use-params"


> nextjs-project@0.0.0 testonly <snip>/next.js
> jest --runInBand "test/e2e/app-dir/use-params"

Using test mode: start in test folder e2e
packageJson?? {}
Creating test directory with isolated next... (use NEXT_SKIP_ISOLATE=1 to opt-out)
Creating next instance in:
/private/var/folders/qs/g2b94nfj79538hg_1y6t2g500000gn/T/next-install-9e8b658884cf0348e5a6db59fb80d2d5041906754abb0b2571364f41d00d50a3
Creating temp repo dir /private/var/folders/qs/g2b94nfj79538hg_1y6t2g500000gn/T/next-repo-ff065aed6eec90e35ce5e2a1813a1ebe6774e4f4a3fe87705ad6fc245410b2a0
{
  swcDirectory: '<snip>/next.js/packages/next-swc/native'
}
  console.log
    using swc dep { nextSwcVersion: undefined, nextSwcPkg: undefined }

      at log (../.github/actions/next-stats-action/src/prepare/repo-setup.js:168:23)


 RUNS  test/e2e/app-dir/use-params/use-params.test.ts
Packages: +35
+++++++++++++++++++++++++++++++++++
Progress: resolved 53, reused 34, downloaded 2, added 35, done

dependencies:
+ @types/node 20.14.0
+ @types/react 18.3.3
+ @types/react-dom 18.3.0
+ next 15.0.0-canary.7
+ react 19.0.0-rc-f994737d14-20240522
+ react-dom 19.0.0-rc-f994737d14-20240522
+ typescript 5.4.5
created next.js install, writing test files
  console.log
    running pnpm next build

      at NextStartInstance.log [as start] (lib/next-modes/next-start.ts:86:13)

  ▲ Next.js 15.0.0-canary.7

   Creating an optimized production build ...
 ✓ Compiled successfully
   Linting and checking validity of types ...


===== TS errors =====

[Test Mode] ./node_modules/.pnpm/file+..+next-repo-ff065aed6eec90e35ce5e2a1813a1ebe6774e4f4a3fe87705ad6fc245410b2a0+packages+n_36co7dw5syrhkh2w56ltmezm4i/node_modules/next/navigation-types/compat/navigation.d.ts:11:19
Type error: Overload signatures must all be exported or non-exported.

   9 |    * ready.
  10 |    */
> 11 |   export function useSearchParams(): ReadonlyURLSearchParams | null
     |                   ^
```